File: benchmarks/struct_records_inout/struct_records_inout.go
package main

import "fmt"

type Person struct {
	Name string
	Age  int
}

func makePerson(i int) Person {
	return Person{Name: fmt.Sprintf("user%d", i), Age: 20 + i%50}
}

func birthday(p *Person) {
	p.Age++
}

func score(p Person) int {
	return len(p.Name) + p.Age
}

func main() {
	total := 0
	for i := range 500000 {
		p := makePerson(i)
		birthday(&p)
		total += score(p)
	}
	if total != 27638890 {
		panic("struct_records_inout checksum")
	}
	fmt.Println("assert passed, struct_records_inout is correct")
}

File: benchmarks/struct_records_inout/struct_records_inout.py
class Person:
    __slots__ = ("age", "name")

    def __init__(self, name, age):
        self.name = name
        self.age = age


def make_person(i):
    return Person("user" + str(i), 20 + i % 50)


def birthday(p):
    p.age += 1


def score(p):
    return len(p.name) + p.age


def main():
    total = 0
    for i in range(500000):
        p = make_person(i)
        birthday(p)
        total += score(p)
    assert total == 27638890, "struct_records_inout checksum"
    print("assert passed, struct_records_inout is correct")


main()
